src/components/Card/Chart/PeriodPostCountCard.jsx

```diff
@@ -0,0 +1,41 @@
+import { useState } from "react";
+
+import asyncGetPostCountList from "../../../api/keyword/asyncGetPostCountList";
+import { PERIOD_TYPE } from "../../../config/constants";
+import { getCursorDate } from "../../../utils/date";
+import LineChart from "../../Chart/LineChart";
+import PeriodPagination from "../../Pagination/PeriodPagination";
+import { keepPreviousData, useQuery } from "@tanstack/react-query";
+import PropTypes from "prop-types";
+
+const PeriodPostCountCard = ({ keywordId }) => {
+  const [cursorId, setCursorId] = useState(() => getCursorDate());
+
+  const { data: chartData, isPlaceholderData } = useQuery({
+    queryKey: ["postCount", keywordId, cursorId],
+    queryFn: () => asyncGetPostCountList(keywordId, cursorId, PERIOD_TYPE.WEEKLY),
+    placeholderData: keepPreviousData,
+  });
+
+  if (typeof chartData === "undefined") {
+    return <div className="mt-100 w-1/3 p-10">조회된 주간 게시물이 없습니다 🥲</div>;
+  }
```

Comment:
지금도 좋으나, 네트워크 fetching 중인 상태에서는 `undefined`일테고, undefined인 상태로 자식 컴포넌트의 props 값으로 내려갈 것으로 생각이 됩니다. (자식 컴포넌트에서는 해당 값이 `isRequired`이구요. undefined가 내려온다면 분명 에러가 생길 것입니다.)

따라서, 아직 데이터가 isPending인 상태에서는 안전장치의 null로 렌더링하고, 주간 게시물이 없는지는 별도로 체크하는 것이 좋아보입니다. 

```suggestion
  if (chartData === undefined) {
    return null;
  }
  
  if (chartData.postCountList.every(count => count === 0)) {
    return <div className="mt-100 w-1/3 p-10">조회된 주간 게시물이 없습니다</div>;
  }
```

Reply:
데이터를 가지고 오는 중과 데이터를 가지고 왔지만 데이터가 없는 경우에 대해서 구분해주셔서 감사해요!
다만, 규경님이 제안해주신 코드와 저의 코드를 비교해보면
주간의 모든 게시물 수가 0인 경우에,
1. 차트를 보여주지만, 차트 데이터는 모두 0으로 표시됨
2. 차트를 보여주지 않고, "조회된 주간 게시물이 없습니다" 문구 표시

어떻게 보여줘야 할지 정해야할 것 같네요! 
cc @jin-ttao

Reply:
페이지네이션 기능이 있어서 저는 1번으로 표기되는게 좋겠다는 의견이에요. 
조회기간에 따라 게시물이 없는 주간이 걸리면 그래프 전체 혹은 일부가 나타났다가 사라지게 될 것으로 보여요.
일관되게 0으로 차트는 계속 노출시켜도 무방하다고 생각이 드네요! 😃

Reply:
생각해보니, `chartData.postCountList.every(count => count === 0)` 인 경우는 서버에서 hasNextCursorId 의 값으로 false를 내려주기에, 버튼 자체가 `disabled` 될 것 같네요. 그리고 더 깊이 생각해보면, 지금 당장은 저희 로직이 서버에서 다음 주나 누적된 전 주의 데이터에 대한 게시물의 수에 대해 hasNext, hasPervious 값을 계산하여 true/false로 내려주지만, 중간이 비어있다가 한참 뒤에 하나, 둘 씩 있는 게시물을 생각하면 약간 아찔하긴 하네요. 

우선은 깊게 들어가지말고, 지금 사안만 따져본다면, 어차피 버튼이 `disabled`될 것이고, 만약 버튼을 `disabled` 하지 않을 것이라면 1번이 낫겠네요. 하지만, 계속 클릭했을 때 데이터가 나온다면 모르겠지만, 나오지 않는다면 무의미한 클릭이 계속 될 것 같네요.

결국 제 생각에는 현재로써는 에러 처리만 하면 될 것 같습니다..!

Reply:
![스크린샷 2024-11-14 오전 10 39 05](https://github.com/user-attachments/assets/2196d13a-a431-4ba2-8e17-348684b4855f)
에러 처리 추가했습니다!